aptos-move/framework/move-stdlib/sources/vector.move

```diff
@@ -260,6 +275,24 @@ module std::vector {
         pragma intrinsic = true;
     }
 
+    /// Remove the first occurrence of a given value in the vector `v` and return it in
+    /// a vector, swapping the last element in the vector with the removed item.
+    /// This is O(n) + O(1) and does NOT preserve ordering of elements in the vector.
+    /// This returns an empty vector if the value isn't present in the vector.
+    /// Note that this cannot return an option as option uses vector and there'd be a
+    /// circular dependency between option and vector.
+    public fun swap_remove_value<Element>(v: &mut vector<Element>, val: &Element): vector<Element> {
```

Comment:
I wonder whether this should be generalized to take a predicate, and whether we really need the 'swap' in the name. I would perhaps just have
```
inline fun find_remove<E>(v: &mut vector<E>, p: |Element|bool) -> vector<E>
```

Reply:
Agreed on naming.